src/V3LinkInc.cpp

```diff
@@ -0,0 +1,239 @@
+        iterateAndNextNull(nodep->bodysp());
+        iterateAndNextNull(nodep->incsp());
+        // Done the loop
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    virtual void visit(AstNodeFor* nodep) VL_OVERRIDE {
+        nodep->v3fatalSrc(
+            "For statements should have been converted to while statements in V3Begin.cpp");
+    }
+    virtual void visit(AstNodeStmt* nodep) VL_OVERRIDE {
+        if (!nodep->isStatement()) {
+            iterateChildren(nodep);
+            return;
+        }
+        m_insMode = IM_BEFORE;
+        m_insStmtp = nodep;
+        iterateChildren(nodep);
+        m_insStmtp = NULL;  // Next thing should be new statement
+    }
+    void unsupported_visit(AstNode* nodep) {
+        m_unsupportedHere = true;
+        iterateChildren(nodep);
+        m_unsupportedHere = false;
+    }
+    virtual void visit(AstLogAnd* nodep) VL_OVERRIDE { unsupported_visit(nodep); }
+    void prepost_visit(AstNode* nodep) {
+        // Check if we are underneath a statement
+        if (m_insStmtp == NULL) {
+            prepost_non_stmt_visit(nodep);
+        } else {
+            prepost_stmt_visit(nodep);
+        }
+    }
+    void prepost_non_stmt_visit(AstNode* nodep) {
+        iterateChildren(nodep);
+
+        AstNRelinker linker;
+        AstNode* innerp = nodep->op1p();
+        AstNode* inner_rhsp = innerp->cloneTree(true);
+        AstConst* constp = VN_CAST(nodep->op2p(), Const);
+        AstConst* newconstp = constp->cloneTree(true);
+
+        // Make sure it can be lvalue if it's a VarRef
+        if (VN_IS(innerp, VarRef)) {
+            AstNodeVarRef* vr = VN_CAST(innerp, VarRef);
+            vr->lvalue(true);
+        }
+
+        innerp->unlinkFrBack(&linker);
+
+        // Prepare an assignment
+        FileLine* fl = nodep->fileline();
+
+        AstAssign* assignp;
+        if (VN_IS(nodep, PreSub) || VN_IS(nodep, PostSub)) {
+            assignp = new AstAssign(fl, innerp, new AstSub(fl, inner_rhsp, newconstp));
+        } else {
+            assignp = new AstAssign(fl, innerp, new AstAdd(fl, inner_rhsp, newconstp));
+        }
+
+        nodep->replaceWith(assignp);
+
+        VL_DO_DANGLING(nodep->deleteTree(), nodep);
+    }
+    void prepost_stmt_visit(AstNode* nodep) {
+        if (m_unsupportedHere) {
+            nodep->v3error("Incrementation is not yet supported in this context.\n");
+        }
+
+        iterateChildren(nodep);
+
+        AstNodeVarRef* vr = VN_CAST(nodep->op1p(), VarRef);
+        AstConst* constp = VN_CAST(nodep->op2p(), Const);
+        AstNode* backp = nodep->backp();
+        AstConst* newconstp = constp->cloneTree(true);
+
+        // Prepare a temporary variable
+        FileLine* fl = backp->fileline();
+        string name = string("__Vincrement") + cvtToStr(m_modIncrementsNum++);
+        AstVar* varp = new AstVar(fl, AstVarType::BLOCKTEMP, name, vr->varp()->subDTypep());
```

Comment:
```suggestion
        AstVar* varp = new AstVar(fl, AstVarType::BLOCKTEMP, name, VFlagChildDType(), vr->varp()->subDTypep()->cloneTree());
```

Otherwise will soon break with queues (something I need to fix on master).

Reply:
This change causes
```
../V3LinkInc.cpp:174:122: error: no matching function for call to 'AstNodeDType::cloneTree()'
  174 |         AstVar* varp = new AstVar(fl, AstVarType::BLOCKTEMP, name, VFlagChildDType(), vr->varp()->subDTypep()->cloneTree());
```

Reply:
Ok, sorry, leave it as is and I'll need to fix in my feature branch.